components/client/src/main/java/com/hotels/styx/client/healthcheck/monitors/ScheduledOriginHealthStatusMonitor.java

```diff
@@ -105,7 +116,14 @@ private void resetListeners(Set<Origin> origins) {
     @Override
     protected CompletableFuture<Void> stopService() {
         this.hostHealthMonitorExecutor.shutdown();
-        return completedFuture(null);
+
+        return runAsync(() -> {
+            try {
+                hostHealthMonitorExecutor.awaitTermination(10, SECONDS);
+            } catch (InterruptedException e) {
+                throw propagate(e);
```

Comment:
Should we set the thread interrupted flag before propagating?
Also IIRC `Throwables.propagate` is deprecated from recent Guava versions.

Reply:
It seems that the ForkJoinPool clears the thread interruption status (in awaitWork) so it's not a huge issue if we don't clear the status, but the path execution is different and we should never discard the interruption status as a principle.
However, maybe we should not use the JoinFork pool for blocking tasks?